<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <widget class="QFrame" name="frame">
      <property name="frameShape">
       <enum>QFrame::Shape::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="1,0">
         <item>
          <widget class="QLabel" name="label">
           <property name="font">
            <font>
             <pointsize>20</pointsize>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Server</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="PB_clearServerText">
           <property name="text">
            <string>Clear</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="PTE_server">
         <property name="font">
          <font>
           <family>D2Coding</family>
          </font>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QLineEdit" name="LE_serverInput">
           <property name="font">
            <font>
             <family>D2Coding</family>
            </font>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="PB_sendToClient">
           <property name="text">
            <string>Send</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frame_2">
      <property name="frameShape">
       <enum>QFrame::Shape::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Shadow::Raised</enum>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4" stretch="1,0">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="font">
            <font>
             <pointsize>20</pointsize>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Client</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignmentFlag::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="PB_clearClientText">
           <property name="text">
            <string>Clear</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QPlainTextEdit" name="PTE_client">
         <property name="font">
          <font>
           <family>D2Coding</family>
          </font>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLineEdit" name="LE_clientInput">
           <property name="font">
            <font>
             <family>D2Coding</family>
            </font>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="PB_sendToServer">
           <property name="text">
            <string>Send</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>PB_sendToClient</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>sendMsgToClient()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>731</x>
     <y>263</y>
    </hint>
    <hint type="destinationlabel">
     <x>664</x>
     <y>294</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PB_sendToServer</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>sendMsgToServer()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>748</x>
     <y>545</y>
    </hint>
    <hint type="destinationlabel">
     <x>677</x>
     <y>593</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PB_clearServerText</sender>
   <signal>clicked()</signal>
   <receiver>PTE_server</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>727</x>
     <y>42</y>
    </hint>
    <hint type="destinationlabel">
     <x>644</x>
     <y>90</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>PB_clearClientText</sender>
   <signal>clicked()</signal>
   <receiver>PTE_client</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>739</x>
     <y>351</y>
    </hint>
    <hint type="destinationlabel">
     <x>575</x>
     <y>374</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LE_serverInput</sender>
   <signal>returnPressed()</signal>
   <receiver>MainWindow</receiver>
   <slot>sendMsgToClient()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>465</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>346</x>
     <y>300</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>LE_clientInput</sender>
   <signal>returnPressed()</signal>
   <receiver>MainWindow</receiver>
   <slot>sendMsgToServer()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>550</x>
     <y>558</y>
    </hint>
    <hint type="destinationlabel">
     <x>425</x>
     <y>594</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>sendMsgToClient()</slot>
  <slot>sendMsgToServer()</slot>
 </slots>
</ui>
